packages/block-library/src/navigation/edit/index.js

```diff
@@ -767,24 +767,11 @@ function Navigation( {
 					currentMenuId={ ref }
 					clientId={ clientId }
 					canUserCreateNavigationMenu={ canUserCreateNavigationMenu }
+					convertClassicMenu={ convertClassicMenu }
+					handleUpdateMenu={ handleUpdateMenu }
 					isResolvingCanUserCreateNavigationMenu={
 						isResolvingCanUserCreateNavigationMenu
 					}
-					onSelectNavigationMenu={ ( menuId ) => {
-						handleUpdateMenu( menuId );
-					} }
-					onSelectClassicMenu={ async ( classicMenu ) => {
-						const navMenu = await convertClassicMenu(
-							classicMenu.id,
-							classicMenu.name,
-							'draft'
-						);
-						if ( navMenu ) {
-							handleUpdateMenu( navMenu.id, {
-								focusNavigationBlock: true,
-							} );
-						}
-					} }
```

Comment:
I'm not sure about this. 

We could just as easily make this a function in the `edit.js` scope and pass it to each instance of the menu selector. That would leave the selector free to be concerned with rendering.

What we have now is an API where some things `handleUpdate` are working on callback pattern. And others are handled internally passing specialist functions in (`convertClassicMenu`).

I preferred the `on*` callback pattern and if we want to DRY up the code we can extract tge `onSelectClassicMenu` to a reusable function.